### PR TITLE
Support most keywords

### PR DESCRIPTION
**Pull request by xsawyerx**
Edit: Finished this PR.

[You can go through the commit messages themselves, as they contain more information.]

This commit adds the facility for adding keywords. Additionally, I've slowly, over these commits, have added more and more keyword support.

I'm not sure yet what to do when there are multiple arguments in which each one can be an expression, theoretically. I figured at first they will only be of type `Value`. Then I thought they might all be `Value` except the last one, which can be an `Expression`. Now I'm not so sure. `perl` isn't always consistent either. :/

Relevant ticket: #5.

My follow-up is to figure out whether we can have multiple expressions as input, which is usually the case for all keywords. I also need to test the compilation of all keywords using the `perl` interpreter to figure out how far it can stretch its lexing of parameters to keywords.

### CODE REVIEW

File: marpa/lib/Guacamole.pm [hunk 1]
```diff
@@ -100,18 +111,279 @@ ArrowDerefCall    ::= CallArgs
...
+
+OpKeywordEndnetentExpr ::= OpKeywordEndnetent
+
+OpKeywordEndprotoentExpr ::= OpKeywordEndprotoent
+
+OpKeywordEndserventExpr ::= OpKeywordEndservent
+
+OpKeywordGetsocknameExpr ::= OpKeywordGetsockname Expression
+
+OpKeywordGetsockoptExpr ::= OpKeywordGetsockopt Value Value Value
```
Comment (vickenty) [inline marpa/lib/Guacamole.pm]: I think arguments can be arbitrary expressions, not just primitive values.

Also probably needs to have a comma to separate arguments.

File: marpa/lib/Guacamole.pm [hunk 2]
```diff
@@ -100,18 +111,279 @@ ArrowDerefCall    ::= CallArgs
 ArrowMethodCall   ::= Ident CallArgs
 ArrowIndirectCall ::= SigilScalar Ident CallArgs
 
+# All keywords
 OpKeyword ::= OpKeywordAbsExpr
```
Comment (vickenty) [inline marpa/lib/Guacamole.pm:115]: This rule does not seem to be used anywhere?

File: marpa/lib/Guacamole.pm [hunk 3]
```diff
@@ -100,18 +111,279 @@ ArrowDerefCall    ::= CallArgs
...
+                      | OpKeywordChroot
+
+OpKeywordCloseExpr ::= OpKeywordClose Expression
+                     | OpKeywordClose
+
+OpKeywordClosedirExpr ::= OpKeywordClosedir Expression
+
+OpKeywordConnectExpr ::= OpKeywordConnect Value Value
+
+OpKeywordContinueExpr ::= OpKeywordContinue Block
```
[the remark this answers is not in the document]
Comment (xsawyerx) [reply marpa/lib/Guacamole.pm:374]: The documentation for `continue` says it supports a block.

```perl
while (...) {

} continue {

}
```

Perhaps we should only parse `continue` after a `while`?

File: marpa/lib/Guacamole.pm (hunk 1 above)
Comment (xsawyerx) [reply marpa/lib/Guacamole.pm]: I fixed this for everything in a later commit.

`OpKeywordGetsockoptExpr ::= OpKeywordGetsockopt Expression OpComma Expression OpComma Expression`

File: marpa/lib/Guacamole.pm (hunk 2 above)
Comment (xsawyerx) [reply marpa/lib/Guacamole.pm:115]: I think this is also fixed now.

File: marpa/lib/Guacamole.pm [hunk 4]
```diff
@@ -100,18 +111,279 @@ ArrowDerefCall    ::= CallArgs
...
+                    | OpKeywordExit
+
+OpKeywordExpExpr ::= OpKeywordExp Expression
+                   | OpKeywordExp
+
+OpKeywordFcExpr ::= OpKeywordFc Expression
+                  | OpKeywordFc
+
+OpKeywordFcntlExpr ::= OpKeywordFcntl Value Value Value
+                     | OpKeywordFcntl Expression
```
[the remark this answers is not in the document]
Comment (xsawyerx) [reply marpa/lib/Guacamole.pm]: Pushed a fix!

File: marpa/lib/Guacamole.pm (hunk 3 above)
[the remark this answers is not in the document]
Comment (vickenty) [reply marpa/lib/Guacamole.pm:374]: Yes, sounds like `continue BLOCK` is only valid after certain loop statements.  Function-like `continue` is an expression, `my $x = continue();` is valid code. Not sure if you want to allow that.

[the remark this answers is not in the document]
Comment (xsawyerx) [reply marpa/lib/Guacamole.pm:374]: I see what you mean now. Yeah, we can't have it as an expression. It makes no sense.

Interesting though. `perl` parses it and throws an error:

```
$ perl -e'$x = continue()'
Can't "continue" outside a when block at -e line 1.
```

**Pull request merged**